uwblueprint/jump-math
# feat: Handle Unverified Users Trying to Log In

## Notion ticket link

[Handle Unverified Users Trying to Log In](https://www.notion.so/uwblueprintexecs/Handle-Unverified-Users-Trying-to-Log-In-223c5520a124475d893b5a8607a26ff7?pvs=4)



## Implementation description
* When unverified teacher tries to login, they are redirected to a new "Verify Your Email" screen
* Note: admins cannot login without being verified - they are auto verified when they set their password

**Changes**
* Return `emailVerified` field in `login` resolver
   * True if email is verified
* Only set cookies and localStorage values if user is verified  
* Add `sendEmailVerificationLink` resolver / mutation
* `UnverifiedUsers` component -> shown if unverified user tries to login


## Steps to test
1. Create a new teacher account - don't verify it
2. Try to log in

Note: if you enter the wrong password, the existing window alert pops up - should we handle this differently, as might be a security hazard?

https://github.com/uwblueprint/jump-math/assets/69697180/41e401eb-c18d-4328-9845-6a5b45b1ff3f


## What should reviewers focus on?
* UI matches Figma
* No breaking changes


## Checklist
